test/Hamiltonians.jl

```diff
@@ -1748,8 +1748,8 @@ end
         @test (a => v) in offdiagonals(c)
     end
 
-    ods = [DVec(od) for od in offdiagonals(c)]
-    ods_product = DVec(addr => diagonal_element(c)) + sum(ods)
+    ods_product = sum([DVec(od) for od in offdiagonals(c)])
```

Comment:
```suggestion
    ods_product = sum(DVec(od) for od in offdiagonals(c))
```
This should work without allocating a vector first, no?

Not really very important here in the tests, of course.